pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -386,6 +402,11 @@ public String alterTableStateOrListTableConfig(
       String endpointUrl = request.getRequestURL().toString();
       AccessControlUtils.validatePermission(tableName, AccessType.UPDATE, httpHeaders, endpointUrl,
           _accessControlFactory.create());
+      // Convert the action to EnableTable, DisableTable and DropTable corresponding to StateType enum.
+      if (!_accessControlFactory.create().hasAccess(httpHeaders, TargetType.TABLE, tableName,
```

Comment:
Suggest doing a switch on `stateType` and use the constant from `Actions` for readability and easier code navigation.